src/modules/ens/ens.service.ts

```diff
@@ -164,8 +166,11 @@ export class EnsService implements OnModuleDestroy {
     });
 
     // Register event handler for owner change or namespace deletion
-    this.ensRegistry.on('NewOwner', async (node, _, owner) => {
-      await this.eventHandler({ hash: node, owner });
+    this.ensRegistry.on('NewOwner', async (node, label, owner) => {
+      await this.eventHandler({
+        hash: solidityKeccak256(['bytes', 'bytes'], [label, node]),
```

Comment:
I don't really get this 😓 . So there was a bug here before? You need to hash both the `label` and the `node` together? I thought that `node` was the namehash of the domain

Reply:
`eventHandler` is expected to be given hash of the transferred domain, which consist from label and node. I think we haven't discovered this bug yet because `NewOwner` event is emitted by `setSubnodeOwner` which we haven't used yet, although this method is preset in iam-lib

Reply:
Hmm, interesting, thanks. Do you think it would make sense to use https://docs.ethers.io/v5/api/utils/hashing/#utils-namehash for this rather than `solidityKeccak256` directly? (sorry if it's a dumb suggestion 😆 )

Reply:
Right, that makes sense 👍  So will we need to trigger this event again for the previously migrated domains? Or will the daily ENS sync fix it you think?

Reply:
I think daily sync should fix this